### PR TITLE
IGNITE-16742 Extend calcite module documentation

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/README.txt [hunk 1]
```diff
@@ -4,7 +4,36 @@ Apache Ignite Calcite Module
...
+Config example:
+
+<bean id="queries.cfg" class="org.apache.ignite.configuration.IgniteConfiguration">
+    <property name="clientConnectorConfiguration">
+        <bean class="org.apache.ignite.configuration.ClientConnectorConfiguration">
+            <property name="host" value="127.0.0.1"/>
+            <property name="port" value="10800"/>
+            <property name="portRange" value="10"/>
+        </bean>
+    </property>
```
Comment (alex-plekhanov) [inline modules/calcite/README.txt:17]: This part looks redundant for SQL configuration documentation

File: modules/calcite/README.txt [hunk 2]
```diff
@@ -13,6 +42,9 @@ When starting a standalone node, move 'optional/ignite-calcite' folder to 'libs'
 Note: At now some logic from ignite-indexing module is reused, this means ignite-indexing module also
 has to be present at classpath.
 
+Connection url via jdbc with calcite engine: jdbc:ignite:thin://127.0.0.1:10800?queryEngine=calcite
```
Comment (alex-plekhanov) [inline modules/calcite/README.txt:45]: Let's add something like `If more than one engine configured ... `
Also let's add about other ways to send query to specific engine (ODBC, hints)
JDBC - should be in upper case (perhaps "Calcite", "H2" - start with upper case too).

File: examples/config/queries-default.xml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:util="http://www.springframework.org/schema/util"
+       xsi:schemaLocation="
+        http://www.springframework.org/schema/beans
+        http://www.springframework.org/schema/beans/spring-beans.xsd
+        http://www.springframework.org/schema/util
+        http://www.springframework.org/schema/util/spring-util.xsd">
+
+    <bean id="queries.cfg" class="org.apache.ignite.configuration.IgniteConfiguration">
```
Comment (alex-plekhanov) [inline examples/config/queries-default.xml:29]: Rename file to `example-sql.xml`?
Inherit configuration from `example-default.xml` and add only `sqlConfiguration`?

File: modules/calcite/README.txt (hunk 1 above)
Comment (zstan) [reply modules/calcite/README.txt:17]: explicit always preferable and more readable that implicit i just want to show dependency between port in config and jdbc url.

**Review by alex-plekhanov — APPROVED** [review]

**Pull request closed without merging**